### PR TITLE
PT-1304 Display inputTime property when available

### PR DESCRIPTION
**Pull request by franck-fourel**
A PR should come in mdblp/tideline to manage the objects with EventIDs: objective will be to deduplicate those objects while keeping the most recent.
This PR is focusing on the display of the inputTime in the PA tooltip.

### CODE REVIEW

File: src/components/daily/bolustooltip/BolusTooltip.js [hunk 1]
```diff
@@ -308,7 +303,7 @@ class BolusTooltip extends React.Component {
     );
     const inputLine = inputTime && (
       <div className={styles.input}>
-        <div className={styles.label}>{t('Entered at {{date}}', {date: this.getFormatedInputTime(bolus, timePrefs)})}</div>
+        <div className={styles.label}>{t('Entered at {{date}}', {date: formatInputTime(bolus.inputTime, timePrefs)})}</div>
```
Comment (yvon-dblg) [inline src/components/daily/bolustooltip/BolusTooltip.js]: `'bolus.inputTime' is missing in props validationeslintreact/prop-types`
Add line 437: `inputTime: PropTypes.string,`
```
  bolus: PropTypes.shape({
    type: PropTypes.string.isRequired,
    inputTime: PropTypes.string,
  }).isRequired,
```

File: src/components/daily/physicaltooltip/PhysicalTooltip.js [hunk 2]
```diff
@@ -66,8 +67,18 @@ class PhysicalTooltip extends React.Component {
...
+        <div key={'inputTime'} className={styles.pa}>
+          <div className={styles.input}>
+            {t('Entered at {{date}}', {date: formatInputTime(pa.inputTime, timePrefs)})}
+          </div>
+        </div>        
+      )
+    }
 
-    return <div className={styles.container}>{rows}</div>;
+  return <div className={styles.container}>{rows}{inputTime}</div>;
```
Comment (yvon-dblg) [inline src/components/daily/physicaltooltip/PhysicalTooltip.js]:
```suggestion
    return <div className={styles.container}>{rows}{inputTime}</div>;
```

File: src/components/daily/physicaltooltip/PhysicalTooltip.js [hunk 3]
```diff
@@ -66,8 +67,18 @@ class PhysicalTooltip extends React.Component {
...
     ];
+    const inputTime = [];
+    if (pa.inputTime) {
+      inputTime.push(
+        <div key={'inputTime'} className={styles.pa}>
+          <div className={styles.input}>
+            {t('Entered at {{date}}', {date: formatInputTime(pa.inputTime, timePrefs)})}
+          </div>
+        </div>        
+      )
```
Comment (yvon-dblg) [inline src/components/daily/physicaltooltip/PhysicalTooltip.js]:
```suggestion
      );
```

File: src/components/daily/physicaltooltip/PhysicalTooltip.js [hunk 4]
```diff
@@ -66,8 +67,18 @@ class PhysicalTooltip extends React.Component {
...
       </div>,
     ];
+    const inputTime = [];
+    if (pa.inputTime) {
+      inputTime.push(
+        <div key={'inputTime'} className={styles.pa}>
+          <div className={styles.input}>
+            {t('Entered at {{date}}', {date: formatInputTime(pa.inputTime, timePrefs)})}
+          </div>
+        </div>
```
Comment (yvon-dblg) [inline src/components/daily/physicaltooltip/PhysicalTooltip.js]:
```suggestion
        </div>
```

File: src/utils/format.js [hunk 5]
```diff
@@ -122,6 +123,16 @@ export function formatPercentage(val, precision = 0) {
   return format(`.${precision}%`)(val);
 }
 
+/**
+ * formatInputTime
+ * @param {Number} val - raw decimal proportion, range of 0.0 to 1.0
+ *
+ * @return {String} percentage
```
Comment (yvon-dblg) [inline src/utils/format.js]:
```suggestion
 * Format Input Time
 * @param {string|number|Date|moment.Moment} utcTime Zulu timestamp (Integer hammertime also OK)
 * @param {{timezoneAware: boolean, timezoneName?: string}} timePrefs
 *
 * @return {string} The formated time for input time in the terminal
```

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
It works 👍 
There is a few minor things to update.

In `src/components/daily/physicaltooltip/PhysicalTooltip.js`: Missing normalTime in propTypes for physicalActivity
```
  physicalActivity: PropTypes.shape({
    normalTime: PropTypes.string.isRequired,
    reportedIntensity: PropTypes.string.isRequired,
    duration: PropTypes.shape({
      units: PropTypes.string.isRequired,
      value: PropTypes.number.isRequired,
    }).isRequired,
  }).isRequired,
```

I think the tooltip is not well balanced.
image
@MarineDBL any advice ?

**Comment by franck-fourel** [comment]
@MarineDBL do you want to request any changes in this PR or should we accept it as it is ?

**Comment by MarineDBL** [comment]
@franck-fourel I gave feedback to @yvon-dblg 
Wording is ok but text should be aligned left for the titles and right for the rest - including the hour.

File: src/components/daily/physicaltooltip/PhysicalTooltip.js [hunk 6]
```diff
@@ -57,17 +58,29 @@ class PhysicalTooltip extends React.Component {
...
     ];
+    const inputTime = [];
+    if (pa.inputTime) {
+      inputTime.push(
+        <div key={'inputTime'} className={styles.pa}>
+          <div className={styles.label}>
+            {t('Entered at')}
+          </div>
+          <div className={styles.value}>
+            {t('{{date}}', {date: formatInputTime(pa.inputTime, timePrefs)})}
```
Comment (yvon-dblg) [inline src/components/daily/physicaltooltip/PhysicalTooltip.js]: We will have a translation string `{{date}}` with nothing to be translated.
```suggestion
            {formatInputTime(pa.inputTime, timePrefs)}
```

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
One thing left

**Review by yvon-dblg — APPROVED** [review]
LGTM

**Pull request merged**